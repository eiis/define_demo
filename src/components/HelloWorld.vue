<script setup lang="ts">
import {computed,ref,useAttrs} from 'vue'
//ts导入类型语法
import type {Ref,UnwrapRef} from 'vue'
import type { CustomAttrs} from './type'

export interface Props {
  count?: number
  labels?: string[]
}

let image:Ref<HTMLImageElement | null> = ref(null)

let a:UnwrapRef<Ref<number>> =(1)


const attrs = useAttrs() as CustomAttrs
console.log(attrs,'attrs');

function click(){
  console.log('click');
  
  attrs.onClick()
}


let b:Ref<number> =ref(1)

console.log(image.value);
console.log(a);
console.log(b);

const src = ref('')
const ilogo ='vue'
const url =`../assets/${ilogo}.svg`
function testDynamicImportMetaUrl(ilogo:string) {
  // console.log(`../assets/${ilogo}.png`,'ilogo');
  // const url = `../assets/${ilogo}.svg`
  const metaUrl = new URL(`../assets/${ilogo}.svg`,import.meta.url).href
  src.value = metaUrl
  // document.querySelector(`.dynamic-import-meta-url-img`).src = metaUrl
  console.log(src.value,'metaUrl');
  // console.log(import.meta.url,'import.meta.url');
}

testDynamicImportMetaUrl(ilogo)

// const metaUrl = new URL(`../assets/${ilogo}.png`, import.meta.url)
// console.log('metaUrl', metaUrl)
// console.log('import.meta.url', import.meta.url)
console.log(new URL('../assets/'+ ilogo +'.svg', import.meta.url).href,'URL');
console.log(new URL(`../assets/${ilogo}.svg`, import.meta.url).href,'URL');
console.log(new URL(url, import.meta.url).href,'URL');
interface ButtonHTMLAttributes {
  disabled?: boolean;
  form?: string;
  formAction?: string;
  formEncType?: string;
  formMethod?: string;
  formNoValidate?: boolean;
  formTarget?: string;
  name?: string;
  count?:number;
  type?: 'submit' | 'reset' | 'button';
  value?: string | string[] | number;
}

type ButtonProps = Pick<ButtonHTMLAttributes, 'type' | 'name' | 'count'>;

const {count} = defineProps<ButtonProps>();

interface MyButtonProps extends ButtonHTMLAttributes {
    variant: 'blue' | 'white'
}

//引用一个本地接口
const obj:MyButtonProps = {
  variant:'blue',
  disabled:true
}

console.log(obj);


// const {count} =defineProps<{ count: number }>()
// const { count } = toRefs(props);
const even = computed(() =>  (count! % 2 === 0 ? 'even' : 'odd'));
</script>

<template>
  <div class="hello">
    <h1 class="count">{{ count }}</h1>
    <h1> The number is {{ even }}</h1>
    <button @click="click">Click</button>
    <!-- <img class="dynamic-import-meta-url-img" src="../assets/vue.svg"/> -->
    <!-- <img class="dynamic-import-meta-url-img" :src="src"/> -->
  </div>
</template>

<style scoped>
.count {
  color: red;
}
</style>
