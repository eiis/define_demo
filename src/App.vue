<script setup lang="ts">
import {ref} from 'vue'
import HelloWorld from "./components/HelloWorld.vue"

const count = ref(2)

function click(){
  count.value +=1
}
</script>

<template>
  <HelloWorld :count="count" />
  <button @click="click">Click</button>
</template>

