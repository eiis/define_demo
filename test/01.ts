// sum.test.js
import { expect, test,describe } from 'vitest'
import { mount } from '@vue/test-utils'
import { it } from 'vitest'
import App from '../src/App.vue'
import HelloWorld from '../src/components/HelloWorld.vue'

function sum(a,b){
  return a + b
}

// const stock = {
//   type: 'apples',
//   count: 11,
// }

test('adds 1 + 2 to equal 3', () => {
  expect(sum(2, 3)).toMatchSnapshot()
  expect(sum(2, 3)).toBe(5)
})

// test('stock has 13 apples', () => {
//   expect(stock.type).toBe('apples')
//   expect(stock.count).toBe(11)
// })


describe('renders HelloWorld component', () => {
  it('renders a div', () => {
    const wrapper = mount(HelloWorld,{props:{count:2}})
    console.log('wrapper', wrapper.text())
    expect(wrapper.attributes('class')).toBe('hello')
    expect(wrapper.text()).toContain('2')
    expect(wrapper.text()).toContain('even')
  })
})

describe('App', () => {
  it('should increment count value when button is clicked', async () => {
    // 挂载组件
    const wrapper = mount(App);
    
    // 检查初始 count 值
    const helloWorldComponent = wrapper.getComponent({ name: 'HelloWorld' });
    expect(helloWorldComponent.props('count')).toBe(1);
    // expect(helloWorldComponent.props('name')).toBe('name');

    
    // 找到并点击按钮
    // await wrapper.find('button').trigger('click');
    
    // 检查 count 值是否增加
    // expect(helloWorldComponent.props('count')).toBe(3);
  });
});
