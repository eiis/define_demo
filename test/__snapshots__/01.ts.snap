// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`adds 1 + 2 to equal 3 1`] = `5`;
